src/main/java/fi/hsl/transitdata/metroats/MetroEstimatesFactory.java

```diff
@@ -121,10 +121,17 @@ public MetroEstimatesFactory(final PulsarApplicationContext context) {
                 metroEstimateBuilder.setDirection(map.get(TransitdataProperties.KEY_DIRECTION));
         });
 
+        Optional<String> directionString = Optional.of(metroJourneyData.get().get(TransitdataProperties.KEY_DIRECTION));
+        if(!directionString.isPresent()) {
+            log.warn("Couldn't read directionString from metroJourneyData: {}.", directionString);
+            return Optional.empty();
+        }
+        Integer direction = Integer.parseInt(directionString.get());
+
         // routeRows
         List<MetroAtsProtos.MetroStopEstimate> metroStopEstimates = new ArrayList<>();
         for (MetroStopEstimate metroStopEstimate : metroEstimate.routeRows) {
-            Optional<MetroAtsProtos.MetroStopEstimate> maybeMetroStopEstimate = toMetroStopEstimate(metroStopEstimate);
+            Optional<MetroAtsProtos.MetroStopEstimate> maybeMetroStopEstimate = toMetroStopEstimate(metroStopEstimate, direction);
             if (!maybeMetroStopEstimate.isPresent()) {
                     return Optional.empty();
```

Comment:
Should we discard all stop estimates if one fails? Dunno.

Reply:
Now we discard all stop estimates for 1 metro. We do the same thing if for example metroEstimate.trainType is missing etc.

I think this is fine, we log it out so if this happens, you can see it from the logs.

Reply:
Also if one metro estimate row is empty, we discard the whole metro estimate (at line 135)